### DISCUSSION TITLE
Problem with optimizer for contact binary model

### DISCUSSION THREAD

**Discussion by Staf-M** [phoebe-project/phoebe2#813] [Questions]
Hi all,

For our project we have to model the light curve and radial velocity data for a binary star system. We are convinced we are dealing with a contact binary system. Everything seems to work fine up until the optimizer. Here it never reaches the optimal solutions, even when increasing the maximum number of iterations. We tried resolving some warnings like setting the 'ld_mode_bol' to 'manual' and 'atm' to 'blackbody' and also tried the differential evolution optimizer after the nelder_mead, but none seem to work.
Anyone know where this problem may come from? In case it helps we added some of our code as an attachment.
![prob1](https://github.com/phoebe-project/phoebe2/assets/78790058/a2ff4b9d-e4a8-47df-a8f7-ef1b494553e8)
![prob2](https://github.com/phoebe-project/phoebe2/assets/78790058/57b59884-5c04-4dbb-b417-ea4663eb6562)

Thanks in advance!

**Comment by kecnry** [comment]
My guess is that you do not have a finite probability at the initial position - likely either caused by zeros in the sigmas, or because the system is failing to compute because it is out of the atmosphere table limits (see #809 and #810 for similar cases).  Please check those two things and if those aren't the issue, please attach the bundle (or email it if the data need to remain private) and we can investigate further.

-Kyle

**Reply by Staf-M** [reply]
Hi Kyle,

Thanks for the quick answer! We will try these suggestions and see of that solves it.

**Reply by Staf-M** [reply]
Hi Kyle,

We found that some of the lnprobabilities have -inf, so we tried using the 'phoebe01' and 'nm_fit' compute with atm='blackbody' as you suggested in another thread, however that doesn't seem to fix this. We added the bundles right before the run_solver in the attachment.
[before_run_solver.zip](https://github.com/phoebe-project/phoebe2/files/13662434/before_run_solver.zip)

**Reply by kecnry** [reply]
The `before_run_solver_nmfit.bundle` has the atm set to "blackbody" in the "phoebe01" options but are referencing "nm_fit" in the solver (which still has `atm='ck2004'`).  If you make sure that you can successfully call `run_compute(compute='whatever_label')` with the same compute label as you have in `b.get_value(qualifier='compute', context='solver')`, then your initial lnprobability should be finite.

Switching to blackbody seems like the easiest way to move forward, but just be aware of the assumptions that is making, ensure your system parameters are physical and make sense, and I'd encourage you to try to eventually switch back to actual atmospheres if at all possible (some contact systems it just isn't possible to use atmosphere tables right now, so there is a chance you'll be stuck with blackbody assumptions).

**Reply by Staf-M** [reply]
Hi Kyle,

We tried switching some of these things and it works now! Thanks a lot! With this we can continue and try to change the atmosphere as a next step if possible.
